includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -162,34 +164,50 @@ private function register_apple_pay_domain( $secret_key = '' ) {
 	}
 
 	/**
-	 * Processes the Apple Pay domain verification.
+	 * Updates the Apple Pay domain verification file.
 	 *
-	 * @since 3.1.0
-	 * @version 3.1.0
+	 * @param bool $force True to create the file if it didn't exist, false for just updating the file if needed.
+	 *
+	 * @version 4.3.0
+	 * @since 4.3.0
 	 */
-	public function process_apple_pay_verification() {
-		try {
+	public function update_verification_file( $force = false ) {
 			$path     = untrailingslashit( $_SERVER['DOCUMENT_ROOT'] );
 			$dir      = '.well-known';
 			$file     = 'apple-developer-merchantid-domain-association';
 			$fullpath = $path . '/' . $dir . '/' . $file;
 
-			if ( $this->apple_pay_domain_set && file_exists( $fullpath ) ) {
+			$existing_contents = @file_get_contents( $fullpath );
+			$new_contents = @file_get_contents( WC_STRIPE_PLUGIN_PATH . '/' . $file );
+			if ( ( ! $existing_contents && ! $force ) || $existing_contents === $new_contents ) {
 				return;
 			}
 
 			if ( ! file_exists( $path . '/' . $dir ) ) {
 				if ( ! @mkdir( $path . '/' . $dir, 0755 ) ) { // @codingStandardsIgnoreLine
-					throw new Exception( __( 'Unable to create domain association folder to domain root.', 'woocommerce-gateway-stripe' ) );
+					WC_Stripe_Logger::log( 'Error: ' . __( 'Unable to create domain association folder to domain root.', 'woocommerce-gateway-stripe' ) );
+					return;
```

Comment:
Does swapping out the `throw` for a `return` change the behaviour here? Before the exception would get caught by the outer catch block and `apple_pay_domain_set` would be set to `no`. Now I think we'll return out of the new `update_verification_file` and carry on with the rest of the code in `process_apple_pay_verification`?

Reply:
Good catch! It definitely changes behaviour. I've fixed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1054/commits/7a206319fc9c01e33ec244b5197e40f6715ac8f3 . I couldn't just make the `update_verification_file` `throw` on error because it's being called from a hook too.

Reply:
> I couldn't just make the update_verification_file throw on error because it's being called from a hook too

Ah yes, I was wondering why but that makes perfect sense.